app/components/common/Image/index.tsx

```diff
@@ -1,24 +1,26 @@
-import React, { useEffect, useRef } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { ImageType } from './types';
 
-const Image = ({ src, alt, className, placeHolder } : ImageType) => {
+const ImageComp = ({ src, alt, className, placeHolder }: ImageType) => {
   const image = useRef<HTMLImageElement>(null);
+  const [isHydrated, setIsHydrated] = useState(false);
+  useEffect(() => {
+    setIsHydrated(true);
+  }, []);
 
   const imageNotFound = () => {
-    if (image.current) {
+    if (image.current && image.current?.naturalWidth === 0) {
       image.current.src = placeHolder;
     }
   };
 
-  useEffect(() => {
-    if (image.current?.naturalWidth === 0) {
-      imageNotFound();
-    }
-  }, [src, image.current]);
+  if (isHydrated) {
+    return <img ref={image} className={className} src={src} alt={alt} onError={imageNotFound} />;
+  }
 
   return (
-    <img ref={image} className={className} src={src} alt={alt} />
+    null
   );
```

Comment:
I think you should not render null. The component should alway return an image. 
The point was to have a placeholder image that is able to cover the area with a relevant picture. if we were to return null, could handle this with CSS colors.